src/shared/constructStacker.ts

```diff
@@ -3,15 +3,38 @@ import type { Stacker, StackerRepositoryUpdateListener } from "../stacker";
 import { GitSourceControl } from "../source-control";
 import { GitHubCollaborationPlatform } from "../collaboration-platform";
 import { ConcreteStacker } from "../stacker";
+import nodegit from "nodegit";
 
-export function constructStacker(
+export async function constructStacker(
   repoPath: string,
   repositoryUpdateListener: StackerRepositoryUpdateListener,
-): Stacker {
+): Promise<Stacker> {
+  const { owner, repo } = await repoPathToOwnerAndRepo(repoPath);
   return new ConcreteStacker(
     repoPath,
     repositoryUpdateListener,
-    new GitHubCollaborationPlatform(repoPath),
+    new GitHubCollaborationPlatform(repoPath, owner, repo),
     (stackerListener) => new GitSourceControl(repoPath, stackerListener),
   );
 }
+
+async function repoPathToOwnerAndRepo(
+  repoPath: string,
+): Promise<{ owner: string; repo: string }> {
+  const repoResult = await nodegit.Repository.open(repoPath);
+  // TODO: Handle remotes that are not named "origin"
+  const remoteResult = await repoResult.getRemote("origin");
+  const remoteUrl = remoteResult.url();
+  // Sample URLs:
+  // "https://github.com/taneliang/stack-attack"
+  // "git@github.com:taneliang/stack-attack.git"
+  // "git@github.com:taneliang/hello.world"
+  const [rawRepo, owner] = remoteUrl
+    .split("/")
+    .flatMap((s) => s.split(":"))
+    .reverse();
+  const repo = rawRepo.endsWith(".git")
+    ? rawRepo.substr(0, rawRepo.length - 4)
+    : rawRepo;
+  return { owner, repo };
+}
```

Comment:
I moved the ```repoPathToOwnerAndRepo(repoPath: string)``` function to ```constructStacker``` class. Then pass the ```repo``` and ```owner``` to the ```GitHubCollaborationPlatform``` constructor.

Reply:
Looks good!